spec.bs

```diff
@@ -497,6 +502,25 @@ The specification adds the following steps to the [=HTTP fetch=] algorithm, befo
 1. If |redeem response result| is a [=network error=], return |issue response result|.
 
 
+Integration with XMLHttpRequest {#xhr}
+====================================
+
+Modifications to Request {#xhr-request}
+---------------------------------------------------------------
+Refer to [[#fetch-request]].
+
+The <dfn><code>setPrivateToken()</code></dfn> method {#xhr-set-private-token}
+---------------------------------------------------------------
+This specification introduces a new [=setPrivateToken()=] that accepts a {{PrivateToken}} object specifying the {{OperationType}} to execute against the request. It must do the following:
+<ol>
+ <li><p>If [=state=] is not "opened", then throw an
+ "{{InvalidStateError!!exception}}" {{DOMException}}.
+ <li> If OperationType is null, abort.
+ <li> If OperationType is "token-request": [=Append private state token issue request headers=] to the request.
+ <li> If OperationType is "token-redemption": [=Append private state token redemption request headers=] to the request.
+ <li> If OperationType is "send-redemption-record": [=Append private state token redemption record headers=] to the request.
+</ol>
```

Comment:
Add a step for if it doesn't match, to abort or throw an error.

Reply:
updated algorithm so it makes more sense